services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +43,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  xQueueSend(thermalMgrQueueHandle, event, 0);
   return ERR_CODE_SUCCESS;
+  
 }
 
 void osHandlerLM75BD(void) {
+  
   /* Implement this function */
+  
+  float *temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, temp);
+  if (*temp > 75.0) {
+    overTemperatureDetected();
+  }
+  else {
+    safeOperatingConditions();
+  } 
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  
+  float *temp;
   while (1) {
-    
+    thermal_mgr_event_t check; 
+    if (xQueueReceive(thermalMgrQueueHandle, pvParameters , 0) == pdTRUE) {
+      if (check.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
```

Comment:
check was never passed into the call to `xQueueReceive` so it will never be updated after your initialization in line 74